volatility3/framework/plugins/windows/thrdscan.py

```diff
@@ -0,0 +1,141 @@
+##
+## plugin for testing addition of threads scan support to poolscanner.py
+##
+import logging
+import datetime
+from typing import Iterable
+
+from volatility3.framework import renderers, interfaces, exceptions
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins.windows import poolscanner
+from volatility3.plugins import timeliner
+
+vollog = logging.getLogger(__name__)
+
+
+class ThrdScan(interfaces.plugins.PluginInterface, timeliner.TimeLinerInterface):
+    """Scans for windows threads."""
+
+    # cuz installed Framework interface version 2
+    _required_framework_version = (2, 0, 0)
```

Comment:
Hey, thanks for your comment! I appreciate your observation regarding the changes made to the _ETHREAD API in this pull request. I understand your concern about the version of the framework and the requirements.

Could you please provide more information about the specific change you are suggesting in relation to the framework version or requirement framework? Since I'm not deeply familiar with the working and versioning conventions on this project, it will help me better understand your suggestion and make any necessary adjustments.

Thank you again for your feedback, and I appreciate your help! (:

Reply:
Hi there, because this introduces new code to the core threads code (changes in the framework, not just to a plugin), any code that makes use of those changes (including yours) needs to be able to tell if they'll be there or not.  It looks like you've only made an additive change, so you only need to bump the minor version (in `constants/__init__.py`) and then set this plugin's `_required_framework_version` to the same value, so that if someone takes this plugin to another version of volatility, it will fail gracefully if the version is too old.  Otherwise it would work fine until it tried to make use of the new feature and throw an exception which the user would have no idea about.  5;)  I hope that explains the changes that would need to be made?

If you'd changed something, or removed something (so that an old plugin stopped working), then you'd need to change the major version number and we'd need to evaluate if it was a worthwhile/necessary change, since all older plugins would stop working and everything would need their version requirements updating.  Hope that clarifies about the version numbers a bit?  I'll see if I can get @iMHLv2 to have a review of this sometime soon...  5:)

Reply:
Hi ikelos,
Thank you for providing such an elaborate explanation. The modification that @digitalisx suggested is much clearer now.
I will certainly make the needed changes as soon as I can. In fact, I might be able to work on it this weekend. 
Appreciate your patience and guidance in this process (:

Reply:
Hi, 
I have made the necessary changes to the versions in the relevant places.
Bumped the `MINOR_VERSION` version from 4 to 5 (in `constants/__init__.py` file), and the `_required_framework_version` of the `thrdscan `plugin to 2.5.2, accordingly.